fedimint-server/src/config/api.rs

```diff
@@ -580,29 +616,42 @@ pub async fn run_server(
     default_params: ConfigGenParamsRequest,
     module_gens: BTreeMap<u16, (ModuleKind, DynServerModuleGen)>,
     registry: ServerModuleGenRegistry,
-) -> ServerHandle {
-    let state = RpcHandlerCtx {
-        rpc_context: Arc::new(ConfigGenApi::new(
-            data_dir,
-            our_connections.clone(),
-            db,
-            default_params,
-            module_gens,
-            registry,
-        )),
-    };
-    let mut rpc_module = RpcModule::new(state);
-
-    attach_endpoints(&mut rpc_module, config_endpoints(), None);
-
-    ServerBuilder::new()
-        .max_connections(10)
-        .ping_interval(Duration::from_secs(10))
-        .build(&our_connections.api_bind.to_string())
-        .await
-        .expect("Could not start API server")
-        .start(rpc_module)
-        .expect("Could not start API server")
+    mut task_group: TaskGroup,
+) {
+    loop {
+        let (api_sender, mut api_receiver) = tokio::sync::mpsc::channel(1);
```

Comment:
`oneshot::channel`?

Reply:
Had some trouble with that since the Api struct holds onto it and `send` consumes the oneshot.